### PR TITLE
Cleanup class.jetpack-gutenberg.php for clarity

### PR DESCRIPTION
**Pull request by roccotripaldi**
Fixes nothing ( save for our future comprehension of the code )

#### Changes proposed in this Pull Request:

* This PR refactors `class.jetpack-gutenberg.php` for developer clarity - There should be no functional changes.

#### Testing instructions:

- Try using the post editor and ensure Jetpack blocks continue to function as before

#### Proposed changelog entry for your changes:

* No changelog required

### CODE REVIEW

File: class.jetpack-gutenberg.php [hunk 1]
```diff
@@ -24,26 +24,56 @@ function jetpack_register_block( $type, $args = array(), $availability = array(
...
-	private static $blocks_index = array();
+	private static $registered_blocks = array();
+
 	/**
 	 * Add a block to the list of blocks to be registered.
 	 *
 	 * @param string $type Slug of the block.
 	 * @param array  $args Arguments that are passed into the register_block_type.
 	 */
 	public static function add_block( $type, $args, $availability ) {
```
Comment (lezama) [inline class.jetpack-gutenberg.php]: we could call this one `register_block`

File: tests/php/test_class.jetpack-gutenberg.php [hunk 2]
```diff
@@ -17,18 +15,206 @@ public function test_jetpack_register_block_registers_a_gutenberg_block() {
...
+		remove_filter( 'jetpack_set_available_blocks', array( $this, 'add_banana_block' ) );
+
+		// Should not exits in the Module is not active
+		$this->assertFalse( $availability['banana']['available'],  'Avaiability is not false' );
+		$this->assertEquals( $availability['banana']['unavailable_reason'], 'unknown', 'Availability Reason is not unknown' );
+	}
+
+	function test_registered_block_supports_callback_availability() {
+		jetpack_register_block( 'coconut', array(), array(
+			'callback' => function() {
```
Comment (oskosk) [inline tests/php/test_class.jetpack-gutenberg.php]: This anonymous function won't work when run on PHP 5.2

**Review by oskosk — DISMISSED** [review]

**Comment by jeherve** [comment]
Noting that #10938 is related, and seems to be taking somewhat of a different approach?

**Comment by roccotripaldi** [comment]
> Noting that #10938 is related, and seems to be taking somewhat of a different approach?

#10938 is related and complimentary to the work here.

**Review by dereksmart — DISMISSED** [review]
Tested all of our blocks - found no regressions 💪

**Review by dereksmart — APPROVED** [review]
I still approve

**Pull request merged**